dotnet/src/Connectors/Connectors.Memory.AzureAISearch/AzureAISearchServiceCollectionExtensions.cs

```diff
@@ -52,7 +54,7 @@ public static IServiceCollection AddAzureAISearchVectorStore(this IServiceCollec
         Verify.NotNull(endpoint);
         Verify.NotNull(tokenCredential);
 
-        services.AddKeyedTransient<IVectorStore>(
+        services.AddKeyedSingleton<IVectorStore>(
```

Comment:
What are the general rules for vector store objects lifecycle? Is there any specific reason why we register it as `Singleton`? Does it depend on each connector functionality or there is an alignment between connectors?

I think it would be great to collect this information in ADR if possible, so it will be helpful for future connector developers to understand which lifecycle to choose in which case.

Reply:
For each connector the developer really needs to check what the characteristics are of the database client sdk.  E.g. if we are using `HttpClient`, we shouldn't use singleton, since we'll be holding on to a reference for the entire application lifecycle, which is an anti-pattern for `HttpClient`.  On the other hand, some SDKs will tell you, that you should use them as a singleton.  So it really will vary by database.